roles/generate_ssh_key_pair/tasks/main.yml

```diff
@@ -4,6 +4,12 @@
     path: "{{ key_pair_dir }}"
     state: absent
 
+- name: "Make sure {{ fetched_dest }} exists"
+  file:
+    path: "{{ fetched_dest }}"
+    state: directory
+    recurse: yes
```

Comment:
Hmm we've been a bit fast an lose with `bastion` and `localhost` throughout the playbooks. Can we still delegate this to `localhost`. While we strongly suggest they should be the same there is no guarantee they are. I'll take a note to check the playbooks over to check that the distinction is correct and write up a contributor note to explain the difference.

Reply:
Ack, will add the delegate

Reply:
@abays thanks for adding that. Coincidentally I brought up the bastion/localhost thing in a separate PR earlier today. 😄